Spanner/src/Result.php

```diff
@@ -107,7 +107,7 @@ class Result implements \IteratorAggregate
     private $stats;
 
     /**
-     * @var Transaction|null
+     * @var TransactionalReadInterface|null
```

Comment:
Is this a breaking change?

Why this change required is, both Snapshot and Transaction may create the Result object and both of them implements `TransactionalReadInterface`

Reply:
This shouldn't be a breaking change as long as Transaction implements the interface.

Reply:
Looks like the breaking change is because a method was added to a public interface. This is indeed considered breaking because if someone implemented a custom TransactionReadInterface class, they'd get a fatal error. This is why we often label our interfaces as `@internal`. In this case, I don't think there's any real use case for someone providing a custom implementation of that interface, so I think the "breaking change" is acceptable. Although we should go ahead and label the interface as internal.

Reply:
Made the interface `@internal`